apm-agent-core/src/main/java/co/elastic/apm/impl/payload/SystemInfo.java

```diff
@@ -58,6 +79,73 @@ static String getNameOfLocalHost() {
         }
     }
 
+    /**
+     * Finding the container ID based on the {@code /proc/self/cgroup} file.
+     * Each line in this file represents a control group hierarchy of the form
+     * <p>
+     * {@code \d+:([^:,]+(?:,[^:,]+)?):(/.*)}
+     * <p>
+     * with the first field representing the hierarchy ID, the second field representing a comma-separated list of the subsystems bound to
+     * the hierarchy, and the last field representing the control group.
+     *
+     * @return container ID parsed from {@code /proc/self/cgroup} file lines, or {@code null} if can't find/read/parse file lines
+     */
+    private static @Nullable String findContainerId() {
+        String containerId = null;
+        try {
+            Path path = FileSystems.getDefault().getPath("/proc/self/cgroup");
+            if (path.toFile().exists()) {
+                List<String> lines = Files.readAllLines(path, StandardCharsets.UTF_8);
+                for (final String line : lines) {
+                    containerId = parseContainerId(line);
+                    if(containerId != null) {
+                        break;
+                    }
+                }
+            }
+        } catch (Throwable e) {
+            logger.warn("Failed to read/parse container ID from '/proc/self/cgroup'");
+        }
+        logger.debug("container ID is: " + containerId);
```

Comment:
Use a placeholder instead of string concatenation

```suggestion
        logger.debug("container ID is: {}", containerId);
```